### PR TITLE
add du command support

### PR DESCRIPTION
**Pull request by balamurugana**
This command works like `du` command in unices i.e. summarize disk
usage of given TARGET recursively.

### CODE REVIEW

File: cmd/du-main.go [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+      $ {{.HelpName}} s3/jazz-songs
+
+   2. Summarize disk usage of 'louis' prefix in 'jazz-songs' bucket recursively.
+      $ {{.HelpName}} -H s3/jazz-songs/louis/
+`,
+}
+
+// Structured message depending on the type of console.
+type duMessage struct {
+	Prefix string `json:"prefix"`
```
Comment (kannappanr) [inline cmd/du-main.go]: We seemed to use `Status      string  `json:"status"`` everrywhere, so in the case of json, I think applications will look for `status` to be `success`

Comment (balamurugana) [reply cmd/du-main.go]: I believe you are asking to add an additional field called `"status"` with value `"success"`

Comment (harshavardhana) [reply cmd/du-main.go]: It is an mc json API behavior for all mc commands, check this with other commands..

This is the common guidance for all commands given by @abperiasamy

Comment (balamurugana) [reply cmd/du-main.go]: Got it. Why [rmMessage](https://github.com/minio/mc/blob/master/cmd/rm-main.go#L120) doesn't have `status`? Bug in `mc` somewhere or am I missing something?

Comment (harshavardhana) [reply cmd/du-main.go]: It is a bug..

File: cmd/du-main.go [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+
+// du specific flags.
+var (
+	duFlags = []cli.Flag{
+		cli.IntFlag{
+			Name:  "max-depth, d",
+			Usage: "print the total for a folder prefix only if it is N or fewer levels below the command line argument",
+		},
+		cli.BoolFlag{
+			Name:  "human-readable, H",
```
Comment (kannappanr) [inline cmd/du-main.go]: Human-readable, can be by default. You can have `-b` to display in bytes. We can also check with AB.

Comment (balamurugana) [reply cmd/du-main.go]: OK. I will do the change once its confirmed.

Comment (kannappanr) [reply cmd/du-main.go]: @balamurugana Can you confirm this change with AB?

**Comment by kannappanr** [comment]
@balamurugana Can you please resolve the conflict? Also, did you talk to AB about whether human readable should be default?

**Comment by kannappanr** [comment]
@balamurugana Can you please resolve the conflicts

**Comment by kannappanr** [comment]
@balamurugana There won't be a flag called `--human-readable, -H `.

The output will be like this
```
$ mc du play/test
7.1 MB	test
```
Also, `--max-depth value, -d` should be `--depth value, -d`

**Comment by balamurugana** [comment]
@kannappanr 

> There won't be a flag called `--human-readable, -H `.
> 
> The output will be like this
> 
> ```
> $ mc du play/test
> 7.1 MB	test
> ```
> 
> Also, `--max-depth value, -d` should be `--depth value, -d`

Done

**Pull request merged**